### PR TITLE
chore: Refactor to implement model plugin - part 3

### PR DESCRIPTION
**Pull request by Fei-Guo**
This change activates all model plugins. 

- Remove all hard coded model name in workspace_controller.go by using the plugin modelregister.
- Fix related UTs 
- Change the model interface for better abstraction
- Change to use Deployment for llama2-7b models.

### CODE REVIEW

File: docker/kaito/Dockerfile [hunk 1]
```diff
@@ -27,7 +28,7 @@ COPY pkg/ pkg/
 # by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
 RUN --mount=type=cache,target=${GOCACHE} \
     --mount=type=cache,id=kaito-controller,sharing=locked,target=/go/pkg/mod \
-    CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} GO111MODULE=on go build -a -o manager cmd/main.go
+    CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} GO111MODULE=on go build -a -o manager cmd/*.go
```
Comment (ishaansehgal99) [inline docker/kaito/Dockerfile:31]: ok I see this is to include
- main.go
- models.go

File: docker/kaito/Dockerfile [hunk 2]
```diff
@@ -16,9 +16,10 @@ RUN \
     go mod download
 
 # Copy the go source
-COPY cmd/main.go cmd/main.go
+COPY cmd/ cmd/
 COPY api/ api/
 COPY pkg/ pkg/
+COPY presets/ presets/
```
Comment (ishaansehgal99) [inline docker/kaito/Dockerfile:22]: ok includes new presets model.go

File: pkg/utils/plugin/plugin.go [hunk 3]
```diff
@@ -43,3 +43,13 @@ func (reg *ModelRegister) MustGet(name string) inference.Model {
 	}
 	panic("model is not registered")
 }
+
+func (reg *ModelRegister) ListModelNames() []string {
```
Comment (ishaansehgal99) [inline pkg/utils/plugin/plugin.go:47]: Is this called anywhere...

File: pkg/inference/interface.go [hunk 4]
```diff
@@ -4,6 +4,5 @@ package inference
 
 type Model interface {
 	GetInferenceParameters() *PresetInferenceParam
-	NeedStatefulSet() bool
-	NeedHeadlessService() bool
+	SupportDistributedInference() bool //If true, the model workload will be a StatefulSet, using the torch elastic runtime framework.
```
Comment (ishaansehgal99) [inline pkg/inference/interface.go:7]: ah nice change

File: pkg/inference/preset-inferences.go [hunk 5]
```diff
@@ -63,43 +63,35 @@ var (
...
-			inferenceObj.TorchRunRdzvParams["rdzv_backend"] = "c10d"
-			inferenceObj.TorchRunRdzvParams["rdzv_endpoint"] =
-				fmt.Sprintf("%s-0.%s-headless.%s.svc.cluster.local:29500", wObj.Name, wObj.Name, wObj.Namespace)
-		}
-	} else if inferenceObj.ModelName == "Falcon" {
-		inferenceObj.TorchRunParams["config_file"] = "config.yaml"
-		inferenceObj.TorchRunParams["num_processes"] = "1"
-		inferenceObj.TorchRunParams["num_machines"] = "1"
-		inferenceObj.TorchRunParams["machine_rank"] = "0"
-		inferenceObj.TorchRunParams["gpu_ids"] = "all"
```
Comment (ishaansehgal99) [inline pkg/inference/preset-inferences.go:94]: oh wait where did all this falcon torchrun params go?

Comment (Fei-Guo) [reply pkg/inference/preset-inferences.go:94]: They are equal to defaults.

File: test/e2e/preset_test.go [hunk 6]
```diff
@@ -350,7 +350,7 @@ var _ = Describe("Workspace Preset", func() {
 
 		validateAssociatedService(workspaceObj)
 
-		validateInferenceResource(workspaceObj, int32(numOfNode), true)
+		validateInferenceResource(workspaceObj, int32(numOfNode), false)
```
Comment (ishaansehgal99) [inline test/e2e/preset_test.go:353]: got it, since 7b is deployment now

File: pkg/utils/plugin/plugin.go (hunk 3 above)
Comment (Fei-Guo) [reply pkg/utils/plugin/plugin.go:47]: I will add the caller later. It may be good to list all in the log when controller starts.

**Review by ishaansehgal99 — APPROVED** [review]

**Pull request merged**